### DISCUSSION TITLE
tcases-api-test: What does it mean when "No package defined for target"?

### DISCUSSION THREAD

**Discussion by AnirudhChaudhary** [Cornutum/tcases#188] [Q&A] [answered]
Hello,

I am trying to work with the cases-api-test command and I am confused about the functionality of the commands and some corresponding tags. Attached below is a picture that is causing me confusion. When I copied the command structure from the documentation on GitHub, it works fine with the path called src/test/java/org/examples but I don't understand what this path means.

When I change this path I get an error like the one that is seen below as well. I get this error when I omit the -o + path as well.

Can you explain? Thanks.
![TCases-Api-Test Error](https://user-images.githubusercontent.com/52807055/123301280-589e7d80-d4d0-11eb-9284-2f4ad4c3d5e1.jpg)

**Comment by kerrykimbrough** [comment, marked as answer]
The generated JUnit test class -- like any Java class -- must declare the package it belongs to. If no package is defined, Tcases reports a failure like the examples above.

So how can you define the package for the generated test class? There are several ways to do that.

1. Define the package directly
    * ```tcases-api-test -p org.examples ...``` 
    * ```mvn tcases:api-test -DtestPackage=org.examples ...```

2. Define the test class using a fully-qualified class name that includes the package 
   * ```tcases-api-test -n org.examples.MyTest ...``` 
   * ```mvn tcases:api-test -DtestName=org.examples.MyTest ...```

3. Define an output directory that implicitly indicates the package according to Maven conventions 
   * ```tcases-api-test -o target/generated/java/org/examples ...```
   * ```mvn tcases:api-test -DoutDir=target/generated/java/org/examples ...```

**Reply by AnirudhChaudhary** [reply]
So the one that worked for me above was the third example that you stated? The implicit indication is the path that I provide?

**Reply by kerrykimbrough** [reply]
Correct
